### PR TITLE
completion: fixes for `systemctl status --property`

### LINKED ISSUE #5137: autocompletion for "systemctl show --property=" is borked
### Submission type

  - [x] Bug report
  - [ ] Request for enhancement (RFE)

### systemd version the issue has been seen with

> systemd 231
+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN

### Used distribution

> Fedora 25

### In case of bug report: Expected behaviour you didn't see

I want to show the `EnvironmentFile` property value of a unit:

```
$ systemctl show nginx.service | grep EnvironmentFile
EnvironmentFile=/usr/local/src/systemd-docker-nginx/nginx.service.conf (ignore_errors=no)
```

The expected output would thus be `/usr/local/src/systemd-docker-nginx/nginx.service.conf`.

### In case of bug report: Unexpected behaviour you saw

```
$ systemctl show -p EnvironmentFile nginx.service
Property EnvironmentFile does not exist.
```

### In case of bug report: Steps to reproduce the problem

1. Create service unit `foo.service`
1. `systemctl edit foo.service`
1. In the editor, add 
   ```
   [Service]
   EnvironmentFile=/usr/local/src/systemd-docker-nginx/nginx.service.conf
   ```
1. Run commands above

### PR DESCRIPTION
**Pull request by yuwata**
This introduces `systemd --dump-bus-properties`, and makes shell completions use its output.

Fixes #5137 and #9041.
\===========
The completion for zsh is not tested. I do not know what is `(M)` or `(f)`. Please test the zsh completion.

### CODE REVIEW

**Comment by yuwata** [comment]
Fixes for #9041 is added.

File: shell-completion/bash/systemctl.in [hunk 1]
```diff
@@ -11,12 +11,7 @@ __systemctl() {
 }
 
 __systemd_properties() {
-        local mode=$1
-        { __systemctl $mode show --all;
-         @rootlibexecdir@/systemd --dump-configuration-items; } |
-        while IFS='=' read -r key value; do
-            [[ $value ]] && echo "$key"
-        done
+        @rootlibexecdir@/systemd --dump-bus-properties
```
Comment (keszybz) [inline shell-completion/bash/systemctl.in:14]: Nice.

**Comment by yuwata** [comment]
Also added bash-completion for `portablectl`.

**Comment by keszybz** [comment]
I tested the zsh part, and it didn't work. I have no idea what that M and f do, but apparently they *are* needed for something. Force pushed with that commit split into two: your original bash part, and a separate commit for zsh. I tested the zsh part, and it seems to work fine now.

**Comment by yuwata** [comment]
Thanks a lot!

**Comment by boucman** [comment]
there is no change to the systemd man page, and I don't see any change to systemd --help

is that on purpose ?

**Comment by keszybz** [comment]
`--dump-configuration-items` is listed in `--help` and in the man page. This suggests that `--dump-properties` should too.

**Comment by yuwata** [comment]
Ah, I forgot to add them. I will update this. Thanks.

**Comment by yuwata** [comment]
Updated.
- man page and help message are also updated.
- a small fix for man page of portablectl is added.

**Comment by keszybz** [comment]
Ubuntu autopkgtsests are unlikely to find anything here, since they don't run completions (and they already passed before). Let's merge.

**Pull request merged**